docs/migration-guide.md

```diff
@@ -0,0 +1,708 @@
+
+**Key Principles:**
+- **Start with Manual strategy** - Prevents automatic promotions during initial setup
+- **Migrate one worker deployment at a time** - Reduces risk and allows learning
+- **Test thoroughly in non-production** - Validate the approach before production migration
+- **Preserve running workflows** - Ensure in-flight workflows complete successfully
+- **Enable automation gradually** - Move to Progressive rollouts only after validation
+
+### Migration Phases
+
+#### Phase 1: Preparation
+1. **Install the controller** in non-production environments
+2. **Update worker code** to support versioning
+3. **Test migration process** with non-critical workers
+4. **Prepare CI/CD pipeline changes** for new deployment method
+
+#### Phase 2: Initial Migration
+1. **Choose lowest-risk worker** to migrate first
+2. **Create `TemporalWorkerDeployment` custom resource** with Manual strategy
+3. **Validate controller management** works correctly
+4. **Update deployment pipeline** for this worker
+
+#### Phase 3: Gradual Rollout
+1. **Migrate remaining workers** one at a time
+2. **Enable Progressive rollouts** for validated workers
+3. **Monitor and tune** rollout configurations
+4. **Train team** on new deployment process
+
+### Recommended Migration Order
+
+1. **Background/batch processing workers** - Lower risk, easier to validate
+2. **Internal service workers** - Limited external impact
+3. **Customer-facing workers** - Highest risk, migrate last with most care
+
+## Step-by-Step Migration
+
+### Step 1: Install the Temporal Worker Controller
+
+```bash
+# Install the controller using Helm
+helm install -n temporal-system --create-namespace \
+  temporal-worker-controller \
+  oci://docker.io/temporalio/temporal-worker-controller
+  
+```
+
+### Step 2: Create TemporalConnection Resources
+
+Define connection parameters to your Temporal server(s):
+
+```yaml
+apiVersion: temporal.io/v1alpha1
+kind: TemporalConnection
+metadata:
+  name: production-temporal
+  namespace: default
+spec:
+  hostPort: "production.abc123.tmprl.cloud:7233"
+  mutualTLSSecret: temporal-cloud-mtls  # If using mTLS
+---
+apiVersion: temporal.io/v1alpha1
+kind: TemporalConnection
+metadata:
+  name: staging-temporal
+  namespace: default
+spec:
+  hostPort: "staging.abc123.tmprl.cloud:7233"
+  mutualTLSSecret: temporal-cloud-mtls
+```
+
+### Step 3: Prepare Your Worker Code
+
+Update your worker initialization code to properly handle versioning:
+
+**Before (Unversioned):**
+```go
+// Worker connects without versioning
+worker := worker.New(client, "my-task-queue", worker.Options{})
+```
+
+**After (Versioned):**
+```go
+// Worker must use the build ID/deployment name from environment
+// These are set on the deployment by the controller
+buildID := os.Getenv("TEMPORAL_WORKER_BUILD_ID")
+deploymentName := os.Getenv("TEMPORAL_DEPLOYMENT_NAME")
+if buildID == "" || deploymentName == "" {
+  // exit with an error
+}
+workerOptions := worker.Options{}
```

Comment:
maybe instead of starting with empty options, we should recommend that users start with the SDK-loaded default options so that the worker has the right connection info
https://github.com/temporalio/temporal-worker-controller/blob/071789e70726ccd88e95c83eaa77536daad6e0a9/internal/demo/util/client.go#L37